### PR TITLE
[SPARK-41904][CONNECT][PYTHON] Fix Function `nth_value` functions output

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
Currently, the implementation of nth_value for connect is not correct.
```
Traceback (most recent call last):
  File "/Users/s.singh/personal/spark-oss/python/pyspark/sql/tests/test_functions.py", line 755, in test_nth_value
    self.assertEqual(tuple(r), ex[: len(r)])
AssertionError: Tuples differ: ('a', 1, 'x', None) != ('a', 1, 'x', 'x')

First differing element 3:
None
'x'

- ('a', 1, 'x', None)
?               ^^^^

+ ('a', 1, 'x', 'x')
? 
```
This PR will fix the issue.

### Why are the changes needed?
Try to fix the issue that `nth_value` functions output is not correct.

### Does this PR introduce _any_ user-facing change?
'No'.
New feature.

### How was this patch tested?
Manual tests.

### CODE REVIEW

File: connector/connect/common/src/main/protobuf/spark/connect/expressions.proto [hunk 1]
```diff
@@ -209,11 +209,15 @@ message Expression {
...
 
+    // (Required) Indicates the function should skip null values in the determination of which
+    // row to use.
+    bool ignore_nulls = 4;
+
     // (Required) Indicate if this is a user defined function.
     //
     // When it is not a user defined function, Connect will use the function name directly.
     // When it is a user defined function, Connect will parse the function name first.
-    bool is_user_defined_function = 4;
```
Comment (grundprinzip) [inline connector/connect/common/src/main/protobuf/spark/connect/expressions.proto:216]: please don't change the ids of the proto in a backwards incompatible way. Simply make ignore nulls the 5th id.

File: python/pyspark/sql/tests/connect/test_parity_functions.py [hunk 2]
```diff
@@ -122,8 +122,6 @@ def test_nested_higher_order_function(self):
     def test_np_scalar_input(self):
         super().test_np_scalar_input()
 
-    # TODO(SPARK-41904): Fix nth_value value
-    @unittest.skip("Fails in Spark Connect, should enable.")
     def test_nth_value(self):
         super().test_nth_value()
```
Comment (HyukjinKwon) [inline python/pyspark/sql/tests/connect/test_parity_functions.py:127]:
```suggestion
```

**Review by HyukjinKwon — APPROVED** [review]
LGTM except Martin's comment above.

**Comment by zhengruifeng** [comment]
@beliefer you may also need to change https://github.com/apache/spark/blob/587c9741b0d06413483c1e9e47e61f4d531f82c4/connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala#L848-L856

**Pull request closed without merging**